visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/BusinessLayer/VisitServiceImpl.java

```diff
@@ -109,4 +117,21 @@ private Mono<VetDTO> validateVetId(String vetId) {
         return vetsClient.getVetByVetId(vetId)
                 .switchIfEmpty(Mono.error(new NotFoundException("No vet was found with vetId: " + vetId)));
     }
+
+    private Mono<VisitRequestDTO> validateVisitRequest(VisitRequestDTO dto) {
+        if (dto.getDescription() == null || dto.getDescription().isBlank()) {
+            return Mono.error(new BadRequestException("Please enter a description for this visit"));
+        } else if (dto.getVisitDate() == null) {
+            return Mono.error(new BadRequestException("Please choose a date for your appointment"));
+        }
+        if(dto.getVisitDate().isBefore(LocalDateTime.now())) {
```

Comment:
[Nitpick]: This should also be prevented in the front-end date picker.

Reply:
same explanation as for the required annotation.